cluster-autoscaler/main.go

```diff
@@ -279,6 +279,7 @@ var (
 	checkCapacityBatchProcessing                 = flag.Bool("check-capacity-batch-processing", false, "Whether to enable batch processing for check capacity requests.")
 	checkCapacityProvisioningRequestMaxBatchSize = flag.Int("check-capacity-provisioning-request-max-batch-size", 10, "Maximum number of provisioning requests to process in a single batch.")
 	checkCapacityProvisioningRequestBatchTimebox = flag.Duration("check-capacity-provisioning-request-batch-timebox", 10*time.Second, "Maximum time to process a batch of provisioning requests.")
+	forceDeleteLongUnregisteredNodes             = flag.Bool("force-delete-unregistered-nodes", false, "Whether to enable force deletion of long unregistered nodes, regardless of the min size of the node group the belong to.")
```

Comment:
If we control this via flag, it will stay a flag forever and most people won't turn it on. Could this be inferred from cloudprovider support for `ForceDeleteNodes` instead?

Reply:
I thought about this, but I think there will be cases when even though the `ForceDeleteNodes` is implemented, the Cluster Autoscaler user will not want to go below the min size. This is not the case for GCE/GKE, but I cannot guarantee that this will be the case for other providers, so I think keeping this as a flag is a better option.

We can revisit this later, but for now I don't think we should enable it without the possibility of disabline it if anything goes wrong.

Reply:
Fair enough, but right now this is opt-in, not opt-out. Perhaps we could set it to `true` by default, call `ForceDeleteNodes` on init and fall back to `false` if the error is `NotImplemented`?

As a side note, I think we could simplify CA a bit if cloudproviders explicitly exposed supported capabilities instead of returning errors from a subset of functions, but that's very out of scope here.

Reply:
That would be a nice workaround for now, but unfortunately it is not possible without a bigger refactor. AutoscalingOptions are initialized before the CloudProvider is created, so we have no way at this point to run `ForceDeleteNodes` to check for `NotImplemented`. We could alternatively do that during execution, but this seems like not-that-great solution.

Let's sync on the idea of exposing supported capabilities on SIG meeting.